lib/install/client/install.go

```diff
@@ -147,3 +159,30 @@ type InstallerStrategy struct {
 	// installer service connection.
 	ConnectTimeout time.Duration
 }
+
+// updateEnviron relays part of the client's environment to service
+func updateEnviron(req *systemservice.NewServiceRequest) {
```

Comment:
Nit: Instead of having this method with a side-effect here, what do you think about making this a part of `NewServiceRequest` API? I.e. it could have a field that list all environment variables (by names) that need to be propagated to the service and then systemservice implementation would basically do what this method does.

In general, I notice that our packages tend to have a lot of these one-off utility methods which seem unrelated to the main "idea" of the package they're in, while instead they could be made a part of the API (where possible) which would make them easier discoverable/reusable and lead to better structured code.

Reply:
My gripe is exactly with the abundance of these "utility methods" that seem out of place everywhere in our code base. As a result, we oftentimes end up with a lot of code duplication b/c a person that needs a certain feature (incl. myself) just can't know there's a random method in a random package that does that. I also think that there's nothing bad about adding some extra functionality to a service, even if it's only gonna be used in a single place yet - makes it easier to discover and reuse in future as I mentioned (and brings more structure to the code).

Anywho, I'm not picking on this particular function, just wanted to share my thoughts with you.